repository_service_tuf_worker/services/storage/awss3.py

```diff
@@ -0,0 +1,155 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+from io import BytesIO
+from typing import List, Optional
+
+import awswrangler
+import boto3
+from botocore.exceptions import ClientError
+from securesystemslib.exceptions import StorageError  # noqa
+from tuf.api.metadata import Metadata, T, Timestamp
+from tuf.api.serialization import DeserializationError
+
+from repository_service_tuf_worker import parse_if_secret
+from repository_service_tuf_worker.interfaces import IStorage, ServiceSettings
+
+
+class AWSS3(IStorage):
+    def __init__(
+        self,
+        bucket: str,
+        access_key: str,
+        secret_key: str,
+        region: str,
+        endpoint_url: Optional[str] = None,
+    ) -> None:
+        self._bucket: str = bucket
+        self._region: str = region
+        self._access_key: str = parse_if_secret(access_key)
+        self._secret_key: str = parse_if_secret(secret_key)
+        self._endpoint_url: Optional[str] = endpoint_url
+
+        self._s3_session = boto3.Session(
+            aws_access_key_id=self._access_key,
+            aws_secret_access_key=self._secret_key,
+            region_name=self._region,
+        )
+
+        self._s3 = self._s3_session.client(
+            "s3",
+            aws_access_key_id=self._access_key,
+            aws_secret_access_key=self._secret_key,
+            region_name=self._region,
+            endpoint_url=self._endpoint_url,
+        )
+
+        self._s3_resource = self._s3_session.resource(
+            "s3",
+            aws_access_key_id=self._access_key,
+            aws_secret_access_key=self._secret_key,
+            region_name=self._region,
+            endpoint_url=self._endpoint_url,
+        )
+
+    @classmethod
+    def configure(cls, settings) -> None:
+        aws_access_key_id = settings.AWSS3_STORAGE_ACCESS_KEY
+        aws_secret_access_key = settings.AWSS3_STORAGE_SECRET_KEY
+        region_name = settings.AWSS3_STORAGE_REGION
+        endpoint_url = settings.get("AWSS3_STORAGE_ENDPOINT_URL")
+        s3_resource = boto3.resource(
+            "s3",
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=region_name,
+            endpoint_url=endpoint_url,
+        )
+        buckets = [bucket.name for bucket in s3_resource.buckets.all()]
+        if settings.AWSS3_STORAGE_BUCKET not in buckets:
+            raise ValueError(
+                f"Bucket '{settings.AWSS3_STORAGE_BUCKET}' not found."
+            )
+
+    @classmethod
+    def settings(cls) -> List[ServiceSettings]:
+        return [
+            ServiceSettings(
+                names=["AWSS3_STORAGE_BUCKET"],
+                argument="bucket",
+                required=True,
+            ),
+            ServiceSettings(
+                names=["AWSS3_STORAGE_ACCESS_KEY"],
```

Comment:
Do we want to mention both `AWSS3` and `STORAGE` as a prefix?
Isn't `AWSS3` not enough?

Reply:
We use it for LOCAL as well. I kept to be consistent.